src/main/java/com/cinema/service/impl/mapper/MovieSessionMapper.java

```diff
@@ -0,0 +1,41 @@
+package com.cinema.service.impl.mapper;
+
+import com.cinema.model.MovieSession;
+import com.cinema.model.dto.MovieSessionRequestDto;
+import com.cinema.model.dto.MovieSessionResponseDto;
+import com.cinema.service.CinemaHallService;
+import com.cinema.service.MovieService;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionMapper {
+    private static final String DATE_TIME_FORMATTER = "dd-MM-yyyy HH:mm:ss";
+    private final CinemaHallService cinemaHallService;
+    private final MovieService movieService;
+
+    public MovieSessionMapper(CinemaHallService cinemaHallService, MovieService movieService) {
+        this.cinemaHallService = cinemaHallService;
+        this.movieService = movieService;
+    }
+
+    public MovieSession mapToMovieSessionFromDto(MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setShowTime(LocalDateTime.parse(movieSessionRequestDto.getShowTime(),
+                DateTimeFormatter.ofPattern(DATE_TIME_FORMATTER)));
+        movieSession.setCinemaHall(cinemaHallService
+                .getById(movieSessionRequestDto.getCinemaHallId()));
+        movieSession.setMovie(movieService.getById(movieSessionRequestDto.getMovieId()));
+        return new MovieSession();
```

Comment:
what is wrong here?

Reply:
I shouldn't return new object - fixed it already but I couldn't push this request to this project - have another one and will send you a new link

Reply:
return new - fixed it